packages/http-specs/specs/payload/pageable/main.tsp

```diff
@@ -52,4 +52,248 @@ namespace ServerDrivenPagination {
 
     @nextLink next?: url;
   };
+
+  @route("/continuationtoken")
+  namespace ContinuationToken {
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request body and response body.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ],
+        "nextToken": "12345"
+      }
+      ```
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected request body:
+      ```json
+      { "token": "12345" }
+      ```
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-body-response-body")
+    @list
+    op reqBodyResBody(@continuationToken token?: string): {
```

Comment:
seems continuationToken parameters/properties should be always optional, a linter to limit that?

Reply:
Not sure whether there is a linter. But https://typespec.io/docs/standard-library/pagination/#pagination and TCGC tests set them as optional, so I keep consistent with TCGC and official doc.

Reply:
since the operation is both used for first page and next page. i believe optional is always.